### DISCUSSION TITLE
Hello

### DISCUSSION THREAD

**Discussion by langningchen** [PythonSmall-Q/CDN#8] [General]
Hi, anyone there?

**Comment by PythonSmall-Q** [comment]
yes

**Comment by PythonSmall-Q** [comment]
你知道怎么写自动化登录吗？我想用GithubActions自动登录我那个bbs

**Comment by langningchen** [comment]
不太行，你不是加了验证码吗？

**Comment by langningchen** [comment]
还有我又双叒叕准备放弃Azure了，因为那个只有1G的内存、1vCPU，太慢了

**Comment by langningchen** [comment]
这种低配还要$20/mo，差不多1800RMB一年

**Comment by langningchen** [comment]
我没这么多钱给这个项目

**Comment by PythonSmall-Q** [comment]
登陆验证关掉了

**Comment by PythonSmall-Q** [comment]
Google告诉我自动机器人登录对我的服务器负荷不大

**Reply by PythonSmall-Q** [reply]
两台服务器轮番负荷不会出问题

**Reply by PythonSmall-Q** [reply]
因为服务器会cache用户密码和账户，1小时没有人以此账户登录会删除cache

**Comment by PythonSmall-Q** [comment]
我有均衡负载的

**Comment by langningchen** [comment]
把请求保存下来，让ChatGPT帮你写

**Comment by langningchen** [comment]
或者我来当ChatGPT

**Comment by langningchen** [comment]
不过我先做一会儿题

**Comment by langningchen** [comment]
你要用什么语言写？C++? PHP? JS? YML? Bash?

**Comment by PythonSmall-Q** [comment]
你当chatgpt吧。openai的东西我不信任（也就是为什么当时我没加入测试

**Comment by PythonSmall-Q** [comment]
都行

**Comment by PythonSmall-Q** [comment]
大佬您方便即可

**Comment by langningchen** [comment]
那就 JS 前端

**Comment by langningchen** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/99b97f49-5cdc-4ebe-aae5-30a3d38127b5)

**Reply by PythonSmall-Q** [reply]
6

**Comment by langningchen** [comment]
我先把C题A了再写

**Comment by PythonSmall-Q** [comment]
好

**Comment by langningchen** [comment]
你这样公开密码……

**Comment by PythonSmall-Q** [comment]
emm你说得对，但是即使别人进去了也没用

**Comment by langningchen** [comment]
现在得到的结果：
访问 https://xmoj.flarum.cloud/ 会在响应中包含 csrf token 
以后每次请求有了这个 token 就简单多了

**Comment by langningchen** [comment]
csrf token 找了半天 QWQ

**Comment by langningchen** [comment]
好像可以登录了

**Comment by langningchen** [comment]
你登陆进去以后干什么？

**Comment by langningchen** [comment]
我可以把登录以后的token输出

**Comment by langningchen** [comment]
@PythonSmall-Q

**Comment by PythonSmall-Q** [comment]
登录之后我也不清楚，要么刷一条qp？

**Comment by PythonSmall-Q** [comment]
证明我还活着？

**Comment by langningchen** [comment]
等一下

**Comment by langningchen** [comment]
我发现有人蓄意攻击XMOJ

**Comment by langningchen** [comment]
就是 朱晨瑞

**Comment by langningchen** [comment]
你看他的昵称
http://www.xmoj.tech/userinfo.php?user=zhuchenrui2

**Comment by langningchen** [comment]
看来他MYSQL学的挺多，不过没有足够多，不知道这个问题已经是几年前的了，现在基本上没有网站有这个漏洞了

**Comment by langningchen** [comment]
> 登录之后我也不清楚，要么刷一条qp？

好的

**Comment by langningchen** [comment]
在哪里刷？

**Comment by PythonSmall-Q** [comment]
我开个帖你等一下

**Comment by PythonSmall-Q** [comment]
https://xmoj.flarum.cloud/d/20-shua-qpzhuan-yong

**Reply by langningchen** [reply]
好的

**Comment by PythonSmall-Q** [comment]
用了贵州的CDN，服务器一台在美国，一台在贵阳

**Comment by langningchen** [comment]
有BUG？

**Comment by langningchen** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/d99639c3-7a5e-410b-bc7c-046ea7594b39)

**Comment by langningchen** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/0f87116a-c6ad-4932-b6d9-e6ea274d7ced)

**Comment by langningchen** [comment]
为什么你能发？

**Comment by langningchen** [comment]
修改可以，但是发布新的好像不行

**Comment by PythonSmall-Q** [comment]
az等一下，我换个账户发否则会炸

**Comment by langningchen** [comment]
发布新的Reply

**Comment by langningchen** [comment]
没事

**Comment by PythonSmall-Q** [comment]
再试试

**Comment by langningchen** [comment]
好的

**Comment by langningchen** [comment]
还是不行

**Comment by langningchen** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/03caaad8-acf5-4220-8757-a482c37594df)

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/70f3bb0a-015a-4c77-9995-8e6c102c47fb)
yandex好恶心

**Comment by langningchen** [comment]
不对，他告诉我不行但是能发出去

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/0c297bec-c712-4332-bcc6-55d6d4490b28)
有了

**Comment by langningchen** [comment]
虽然说“请求的资源不存在”，但是发出去了

**Comment by langningchen** [comment]
帮个忙

**Comment by langningchen** [comment]
调个程序好吧？

**Comment by langningchen** [comment]
[这道题](http://www.xmoj.tech/problem.php?cid=6263&pid=3)
```cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
const ll N = 10005;
const ll INF = 0x3FFF'FFFF'FFFF'FFFF;
ll n, m, k, Answer, a[N], b[N], f[N][N];
int main()
{
    scanf("%lld%lld%lld", &n, &m, &k);
    for (ll i = 1; i <= n; i++)
        scanf("%lld%lld", &a[i], &b[i]);
    for (ll i = 1; i <= n; i++)
        for (ll j = 1; j <= k; j++)
            f[i][j] = -INF;
    f[1][0] = a[1];
    Answer = a[1];
    for (ll i = 2; i <= n; i++)
        for (ll j = 1; j <= k; j++)
            for (ll l = 1; l < i; l++)
                if (b[i] - b[l] <= m)
                {
                    f[i][j] = max(f[i][j], f[l][j - 1] + a[i]);
                    Answer = max(Answer, f[i][j]);
                }
    printf("%lld\n", Answer);
    return 0;
}
```

**Reply by PythonSmall-Q** [reply]
#include<bits/stdc++.h>
using namespace std;
const long long N=10009;
long long n,m,d,a[N],b[N],q[N],l,r,f[3][N],ans;
int main(){
	cin>>n>>d>>m;
	for(long long i=1;i<=n;i++){
		cin>>a[i]>>b[i];
	}
	f[0][1]=a[1];
	for(long long i=1;i<n;i++){
		if(b[i+1]-b[i]>d){
			n=i;
			break;
		}
	}
	m=min(m,n-1);
	for(long long k=1;k<=m;k++){
		l=1,r=1;
		q[l]=k;
		long long now=(k&1);
		long long pre=(now^1);
		for(long long i=k+1;i<=n;i++){
			while(l<=r&&b[i]-b[q[l]]>d){
				++l;
			}
			f[now][i]=max(f[pre][q[l]]+a[i],f[now][i]);
			while(l<=r&&f[pre][i]>=f[pre][q[r]]){
				--r;
			}
			q[++r]=i;
		}
	}
	for(long long i=1;i<=n;i++){
		ans=max(ans,f[(m&1)][i]);
	}
	cout<<ans<<endl;
	return 0;
}

**Comment by langningchen** [comment]
为啥RE？本地没问题

**Comment by PythonSmall-Q** [comment]
6我刚看到

**Comment by langningchen** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/49167c18-a328-4b49-b006-d7839cb755bf)
开始刷了

**Comment by langningchen** [comment]
尽管404，不过只要不是400就发出去了

**Comment by langningchen** [comment]
我让他循环了100次
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/75daa60e-5e70-485b-8a6f-ee802617bd51)

**Comment by langningchen** [comment]
额，碰到了这种429……
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/8caa522f-df41-4ba1-982e-2a184ec4551f)

**Comment by PythonSmall-Q** [comment]
诶诶别刷这么多，我看到服务器负荷上去了

**Comment by PythonSmall-Q** [comment]
我前端两个，但是数据库只有一个班

**Reply by PythonSmall-Q** [reply]
半

**Comment by langningchen** [comment]
……

**Comment by langningchen** [comment]
好的，那我就不管他了

**Reply by PythonSmall-Q** [reply]
我手动和谐了

**Comment by langningchen** [comment]
我看到了，出现了不是404的201

**Comment by langningchen** [comment]
你不是说什么语言都可以吗？我用的JS

**Comment by langningchen** [comment]
但是GithubActions没法直接运行JS

**Comment by langningchen** [comment]
只能运行yml

**Comment by PythonSmall-Q** [comment]
az操

**Comment by PythonSmall-Q** [comment]
wc我爱死github了

**Comment by langningchen** [comment]
> 但是GithubActions没法直接运行JS

不过我没有说不能间接运行JS

**Comment by PythonSmall-Q** [comment]
啊啊啊 大佬

**Comment by langningchen** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/026b8d3c-3d66-4be5-a2a4-ead157876d61)
有Image

**Comment by PythonSmall-Q** [comment]
（膜拜

**Reply by PythonSmall-Q** [reply]
等一下人格分裂了

**Comment by langningchen** [comment]
删了也没有用

**Reply by PythonSmall-Q** [reply]
emm

**Comment by langningchen** [comment]
交付几块钱？

**Comment by PythonSmall-Q** [comment]
10？

**Comment by PythonSmall-Q** [comment]
20？

**Comment by langningchen** [comment]
好的，一言为定，20
![image](https://github.com/PythonSmall-Q/CDN/assets/81672884/693ba0a5-7410-4fb4-aab8-f82d1d05646b)

**Reply by PythonSmall-Q** [reply]
啊啊啊？你是不是F12了

**Comment by langningchen** [comment]
不管有没有，反正就20

**Reply by PythonSmall-Q** [reply]
好

**Reply by PythonSmall-Q** [reply]
放学了给你

**Reply by langningchen** [reply]
现在给

**Reply by PythonSmall-Q** [reply]
嗯

**Comment by langningchen** [comment]
PayPal没有那就中午给

**Reply by PythonSmall-Q** [reply]
好

**Reply by PythonSmall-Q** [reply]
包部署和售后吗？

**Reply by langningchen** [reply]
10元质保半年，怎么样？

**Reply by langningchen** [reply]
要么给你一个优惠，质保一年

**Reply by PythonSmall-Q** [reply]
emm人家都是包三年的您包半年真的膜拜！！！！！！aayayayayyay

**Reply by langningchen** [reply]
还是这样吧，质保1天，如果要延保，10元/年

**Comment by langningchen** [comment]
如果你能帮我3次免费调程序，就免费给你质保3年，第一道程序：
https://github.com/PythonSmall-Q/CDN/discussions/8#discussioncomment-6706710

**Reply by langningchen** [reply]
调好一道程序延一年

**Reply by langningchen** [reply]
质保上限到2038/01/19 03:14:07 UTC

**Reply by PythonSmall-Q** [reply]
《2038》那年我都不知道这网站还活着么

**Reply by langningchen** [reply]
你知道2038危机吗？

**Reply by PythonSmall-Q** [reply]
不知道

**Reply by langningchen** [reply]
查查去

**Comment by PythonSmall-Q** [comment]
啥？

**Comment by langningchen** [comment]
导入了讨论，不过有些BUG

**Comment by langningchen** [comment]
就是回复导入可能出了问题

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/96650663-5095-49a4-bc6a-244148599ca7)
炸了

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/17366147-748a-4f7f-8f60-b08b9311ded9)
炸

**Reply by PythonSmall-Q** [reply]
@langningchen

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/404ab928-5c1a-4999-9238-ef2b03ff9bc1)
PHP child开小了

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/c44ae31f-701d-4cbf-bee3-196f693bb6a4)

**Reply by PythonSmall-Q** [reply]
microsoft认证好烦

**Comment by PythonSmall-Q** [comment]
![image](https://github.com/PythonSmall-Q/CDN/assets/106425289/8b170b95-cd1a-4cf5-a30e-e32fd0f2e885)
切换机器人邮箱！ 这个邮箱是无人值守但是我配置了重定向
